cloudinit/sources/DataSourceHetzner.py

```diff
@@ -75,8 +81,18 @@ def get_data(self):
         self.metadata['public-keys'] = md.get('public-keys', None)
         self.vendordata_raw = md.get("vendor_data", None)
 
+        # instance-id and serial from SMBIOS should be identical
+        if self.metadata['instance-id'] != serial:
+            msg = "SMBIOS serial does not match instance ID from metadata"
+            LOG.critical(msg)
+            raise RuntimeError(msg)
```

Comment:
just one nit. just raise the exception, no need to log and raise. i know some places in cloud-init do that, but it really is unnecessary. cloud-init will fail if the exception is raised here (and it will get that message to the log). no additional copy of it is necessary.